index.html

```diff
@@ -40,5 +61,361 @@ <h2>
+        using a permission prompt instead of presenting the user with a picker.
+        For security reasons, it also only works from "cross-origin isolated"
+        documents that opt-in with a document-policy.
+      </p>
+      <section>
+        <h2>
+          <dfn>MediaDevices</dfn> Additions
+        </h2>
+        <pre class="idl"
+>partial interface MediaDevices {
+  Promise&lt;MediaStream&gt; getViewportMedia(optional DisplayMediaStreamConstraints constraints = {});
+};</pre>
+        <dl data-link-for="MediaDevices" data-dfn-for="MediaDevices" class="methods">
+          <dt>
+            <dfn>getViewportMedia</dfn>
+          </dt>
+          <dd>
+            <p>Prompts the user for permission to live-capture the viewport (current tab).</p>
+            <p>
+              The user agent MUST apply any provided <var>constraints</var> to the produced
+              media after permission has been granted.
+            </p>
+            <p>
+              In the case of audio, the user agent MAY present the end-user with an option to
+              include audio from the current viewport in the capture, if available. Like
+              {{MediaDevices/getDisplayMedia()}} with regards to audio+video, the user agent is
+              allowed to not return audio even if the audio constraint is present. If the user
+              agent knows no audio will be shared for the lifetime of the stream it MUST NOT
+              include an audio track in the resulting stream. The user agent MAY accept a
+              request for audio and video by only returning a video track in the resulting
+              stream, or it MAY accept the request by returning both an audio track and a video
+              track in the resulting stream. The user agent MUST reject audio-only requests.
+            </p>
+            <p>
+              Like {{MediaDevices/getDisplayMedia()}}, the {{PermissionState/"granted"}}
+              permission cannot be persisted.
+            </p>
+            <p>When the {{MediaDevices/getViewportMedia()}}
+            method is called, the user agent MUST run the following
+            steps:</p>
+            <ol>
+              <li>
+                <p>If the <a>current settings object</a>'s
+                [=environment settings object/cross-origin isolated capability=]
+                is false, return a promise <a>rejected</a>
+                with a {{DOMException}} object whose {{DOMException/name}}
+                attribute has the value {{SecurityError}}.</p>
+              </li>
+              <li>
+                If the [=relevant settings object=]'s
+                [=environment settings object/responsible document=]'s
+                [=top-level browsing context=]'s
+                <a href="https://wicg.github.io/document-policy/#required-document-policy">
+                required document policy</a> does not contain
+                `Require-Document-Policy: html-capture` (TODO: use correct algorithm),
```

Comment:
we need to define in this spec the said [document policy "configuration point"](https://wicg.github.io/document-policy/#configuration-point)

Reply:
Thanks! I'll look into it.

Reply:
I'll address this in a followup.